raft/fsm.cc

```diff
@@ -583,6 +585,25 @@ void fsm::tick() {
             _current_term, _last_election_time, _clock.now());
         become_candidate(_config.enable_prevoting);
     }
+
+    if (is_follower() && !current_leader() && _ping_leader) {
+        // We are a follower but a leader is not known. It will not be known
+        // until a communication from a leader which (for an idle leader) may
+        // not happen any time soon since we use external failure detector and
+        // our leader does not send periodic empty append messages. By sending
+        // a special append message reject we solicit a reply from a leader.
+        // Non leaders will ignore the append reply.
+        auto& cfg = get_configuration();
+        // If conf is joint it means a leader will send us a non joint one eventually
```

Comment:
I also don't see a reason
@gleb-cloudius

Reply:
If the config is non-joint, the problem is impossible. The idea is to not modify Raft stock behavior so the patch tries to be minimal.

Reply:
> If the config is non-joint, the problem is impossible

You meant when it's joint.
Yes, then it's not possible because the cluster is not idle (the leader is trying to send us a non-joint config), as the comment explains.

However, the patch would be even smaller (in terms of code, not in terms of "behavior") if we didn't special-case joint configs, and it would still work.

Reply:
Ok, I think it's fine as it is too.